### PR TITLE
Fix for Empty Response Body and 0 Content Lenght

### LINKED ISSUE #203: Version 3.1.0 always claims that response body is empty
## :loudspeaker: Describe the bug
After upgrading chucker to version 3.1.0 every single response shows 0 byte size and null body. In app everything works as expected. Downgrading to 3.0.1 fixes the issue

## :bomb: To Reproduce
Do some api request. Go to chucker and click on it. Request is properly shown, but response body is "null". Response size is always 0 bytes"

### :wrench: Expected behavior
Body should be always properly shown.

## :camera: Screenshots
image
image

## :iphone: Smartphone
 - Device: Android Emulator
 - OS: Android 9
 - Lib version: 3.1.0

## :page_facing_up: Additional context
Whole request copied as text:
```
URL: <redacted>
Method: GET
Protocol: http/1.1
Status: Complete
Response: 200 OK
SSL: Yes

Request time: Fri Jan 24 10:58:58 GMT+01:00 2020
Response time: Fri Jan 24 10:58:58 GMT+01:00 2020
Duration: 67 ms

Request size: 0 B
Response size: 0 B
Total size: 0 B

---------- Request ----------

appInstallId: cc5a34e6-9ccf-46d4-9297-04f74a18355e
User-Agent: Android: 1.0.7-rc-dbg-dev (5) ; debug ; debuggableDev
Device-Name: AOSP On IA Emulator
HMAC: 6a2b11
Authorization: Bearer <token here>



---------- Response ----------

content-type: application/json; charset=utf-8
vary: Accept-Encoding
server: envoy
api-supported-versions: 1.0
date: Fri, 24 Jan 2020 09:58:56 GMT
x-envoy-upstream-service-time: 22

null
```

My chucker setup is fairly typical:
```kotlin
// setup collector
val chuckerCollector = ChuckerCollector(
            context = applicationContext,
            showNotification = false,
            retentionPeriod = RetentionManager.Period.ONE_DAY
        )

// adding interceptor
builder.addInterceptor(
            ChuckerInterceptor(
                context,
                collector = chuckerCollector
            )
        )
```

I have quite a few interceptors along the way. ChuckerInterceptor is almost last in the chain, after it there is only `RetryInterceptor`

### PR DESCRIPTION
**Pull request by cortinico**
Fixes a bug introduce by me in #198, when implementing clone of the `okio.Buffer` from the HTTP Response.

Looks like the problem on our end what that we were using `ChuckerInterceptor` as first interceptors, so the Sample App showed valid results. Switching order of the interceptors showed that responses are `null` and the approach followed for the `.clone()` was not correct. I reimplemented it following the `HttpLoggingInterceptor` body.

# How to test

Remove the fix I applied on `ChuckerInterceptor` and keep only the changes I applied in the sample app. It should break and show `null` in the response body.

Fixes #203
Closes #205

### CODE REVIEW

File: sample/src/main/res/xml/network_security_config.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<network-security-config>
+    <debug-overrides>
+        <trust-anchors>
```
Comment (vbuberen) [inline sample/src/main/res/xml/network_security_config.xml]: Why did you add this config into the sample?

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt [hunk 2]
```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
...
+    private fun processResponseBody(response: Response, responseBody: ResponseBody, transaction: HttpTransaction) {
+        val contentType = responseBody.contentType()
+        val charset: Charset = contentType?.charset(UTF8) ?: UTF8
+        val contentLength = responseBody.contentLength()
+
+        val source = responseBody.source()
+        source.request(Long.MAX_VALUE) // Buffer the entire body.
+        var buffer = source.buffer()
+
+        if (CONTENT_ENCODING_GZIP.equals(response.headers()[CONTENT_ENCODING], ignoreCase = true)) {
```
Comment (vbuberen) [inline library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: We already have this check in as `io.bodyIsGzipped()`. You could use it here.
However, I like you approach with constant. Could I ask you to modify `IOUtils` class as well to follow the same approach as here?

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt [hunk 3]
```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
...
+            GzipSource(buffer.clone()).use { gzippedResponseBody ->
```
Comment (vbuberen) [inline library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: Maybe we should move it into `IOUtils`? We have `getNativeSource()` there, which currently used only for request handling. I would move this code there, so both request and response use the same approach.

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt [hunk 4]
```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
...
-            val buffer = source.buffer()
-            var charset: Charset = UTF8
-            val contentType = responseBody?.contentType()
-            if (contentType != null) {
-                try {
-                    charset = contentType.charset(UTF8) ?: UTF8
-                } catch (e: UnsupportedCharsetException) {
-                    return
-                }
+    private fun processResponseBody(response: Response, responseBody: ResponseBody, transaction: HttpTransaction) {
```
Comment (vbuberen) [inline library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: Since you started changes here, will drop a small concern.
I don't understand why can't we just pass only `response` here and retrieve `responseBody` inside this fun? 

Or another possible change - if we look for `response` usages we might see that is used only for headers check. So, maybe it is worth to pass just headers into this fun?

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt [hunk 5]
```diff
@@ -137,32 +134,34 @@ class ChuckerInterceptor @JvmOverloads constructor(
...
+            if (contentLength != 0L) {
+                transaction.responseBody = buffer.clone().readString(charset)
+            }
+        } else {
+            transaction.isResponseBodyPlainText = false
 
-                if (isImageContentType && buffer.size() < MAX_BLOB_SIZE) {
-                    transaction.responseImageData = buffer.readByteArray()
-                }
+            val isImageContentType = (transaction.responseContentType?.contains(CONTENT_TYPE_IMAGE) == true)
```
Comment (vbuberen) [inline library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: Headers are case-insensitive. I know that it is hard to find an API, which sends `Image` or somehow else, except `image`, but let's follow the standard.
https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt (hunk 4 above)
Comment (cortinico) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: > I don't understand why can't we just pass only response here and retrieve responseBody inside this fun?

Yeah we could. Should we make this change in another PR?

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt (hunk 3 above)
Comment (vbuberen) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: Works for me 🆗

File: library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt (hunk 5 above)
Comment (vbuberen) [reply library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt]: Yeah, but it is just one word to add here.
```suggestion
            val isImageContentType = (transaction.responseContentType?.contains(CONTENT_TYPE_IMAGE, ignoreCase = true) == true)
```
